src/smp/ipi.c

```diff
@@ -41,18 +41,17 @@ void ipiStallCoreCallback(bool_t irqPath)
         NODE_STATE(ksSchedulerAction) = SchedulerAction_ResumeCurrentThread;
 
         /* Let the cpu requesting this IPI to continue while we waiting on lock */
-        big_kernel_lock.node_owners[getCurrentCPUIndex()].ipi = 0;
+        big_kernel_lock.node[getCurrentCPUIndex()].ipi = 0;
 #ifdef CONFIG_ARCH_RISCV
         ipi_clear_irq(irq_remote_call_ipi);
 #endif
-        ipi_wait(totalCoreBarrier);
+        ipi_wait();
 
         /* Continue waiting on lock */
-        while (big_kernel_lock.node_owners[getCurrentCPUIndex()].next->value != CLHState_Granted) {
+        while (clh_is_self_in_queue()) {
```

Comment:
This isn't the same check. Previously the loop is checking if the next lock holder has granted (released).  Now it's just checking whether the current node holder is waiting. clh_is_self_in_queue() returns true if the current node is in the queue for the lock and also returns true if it is the current lock holder.  It only returns false when the current node has released the lock and hasn't entered the queue yet.

Reply:
Indeed it isn't, I was confused by the renames and by the comment, `clh_is_self_in_queue` is more subtle than I expected. Fixed.